From 2af552d603cab821a13ea34ffe4efbc6cae6a735 Mon Sep 17 00:00:00 2001
From: Joey <joey@lineageos.org>
Date: Thu, 8 Mar 2018 16:32:08 +0100
Subject: [PATCH 24/46] SystemUI: allow the power menu to be relocated

For devices with power buttons that are not on the right side of the device

Change-Id: Ibb6dbada536e02a02709492c1e37ff7aa5ee3067
Signed-off-by: Joey <joey@lineageos.org>
---
 packages/SystemUI/res/layout/global_actions_wrapped.xml   | 2 +-
 packages/SystemUI/res/values/lineage_config.xml           | 8 ++++++++
 .../src/com/android/systemui/HardwareUiLayout.java        | 1 +
 3 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/res/layout/global_actions_wrapped.xml b/packages/SystemUI/res/layout/global_actions_wrapped.xml
index b715def7329..36d38ee0383 100644
--- a/packages/SystemUI/res/layout/global_actions_wrapped.xml
+++ b/packages/SystemUI/res/layout/global_actions_wrapped.xml
@@ -14,7 +14,7 @@
         android:id="@android:id/list"
         android:layout_width="@dimen/global_actions_panel_width"
         android:layout_height="wrap_content"
-        android:layout_gravity="top|right"
+        android:layout_gravity="@integer/hardware_ui_align"
         android:gravity="center"
         android:orientation="vertical"
         android:padding="12dp"
diff --git a/packages/SystemUI/res/values/lineage_config.xml b/packages/SystemUI/res/values/lineage_config.xml
index 133b0850053..5de08f8b1ce 100644
--- a/packages/SystemUI/res/values/lineage_config.xml
+++ b/packages/SystemUI/res/values/lineage_config.xml
@@ -23,4 +23,12 @@
     <!-- Max visible notification icons -->
     <integer name="config_maxVisibleNotificationIcons">4</integer>
     <integer name="config_maxVisibleNotificationIconsWhenDark">5</integer>
+
+    <!-- Hardware ui (power menu) alignement:
+         Use one of the following values basing on the power button position:
+         * Top | Back -> 49 (Gravity.TOP | Gravity.CENTER_HORIZONTAL)
+         * Left -> 51 (Gravity.TOP | Gravity.LEFT)
+         * Right -> 53 (Gravity.TOP | Gravity.RIGHT)
+    -->
+    <integer name="hardware_ui_align">53</integer>
 </resources>
diff --git a/packages/SystemUI/src/com/android/systemui/HardwareUiLayout.java b/packages/SystemUI/src/com/android/systemui/HardwareUiLayout.java
index 98dc3219acc..74a427eeafc 100644
--- a/packages/SystemUI/src/com/android/systemui/HardwareUiLayout.java
+++ b/packages/SystemUI/src/com/android/systemui/HardwareUiLayout.java
@@ -107,6 +107,7 @@ public class HardwareUiLayout extends FrameLayout implements Tunable {
                 params.bottomMargin = edge;
             } else {
                 params.rightMargin = edge;
+                params.leftMargin = edge;
             }
             mChild.setLayoutParams(params);
         }
-- 
2.17.1

